PyBambooHR/PyBambooHR.py

```diff
@@ -245,6 +244,35 @@ def _format_report_xml(
         xml = '''<report output="{0}">\n\t<title>{1}</title>\n\t{2}<fields>\n{3}\t</fields>\n</report>'''.format(report_format, title, xml_filters, xml_fields)
         return xml
 
+    def _format_time_off_xml(self, request_data):
+        """
+        Utility method for turning dictionary of time_off request data into valid xml.
+        https://www.bamboohr.com/api/documentation/time_off.php#addRequest
+        @param request_data: Dictionary containing incoming data for time off request
+        """
+        fields = ['status', 'start', 'end', 'timeOffTypeId', 'amount']
+        xml = ''
+        for f in fields:
+            value = request_data.get(f)
+            if value:
+                xml += '\n\t<{0}>{1}</{0}>'.format(f, value)
+
+        if request_data.get('notes') and len(request_data.get('notes')) > 0:
+            notes = ''
+            for n in request_data['notes']:
+                f = n.get('type', 'employee')
+                t = n.get('text', '')
+                notes += '\n\t\t<note from="{0}">{1}</note>'.format(f, t)
+            xml += '\n\t<notes>{0}\n\t</notes>'.format(notes)
```

Comment:
```suggestion
        notes = ''
        for note in request_data.get('notes', []):
            f = note.get('type', 'employee')
            t = note.get('text', '')
            notes += '\n\t\t<note from="{0}">{1}</note>'.format(f, t)
            xml += '\n\t<notes>{0}\n\t</notes>'.format(notes)
```

Same pattern can be used for dates :wink: